src/zap.ts

```diff
@@ -249,11 +248,10 @@ export class Zap {
     inputTokenAccountIx && preInstructions.push(inputTokenAccountIx);
     outputTokenAccountIx && preInstructions.push(outputTokenAccountIx);
 
-    const preUserTokenBalance = await getTokenAccountBalance(
-      this.connection,
-      inputTokenAccount,
-      inputTokenAccountIx
-    );
+    const ataExists = !inputTokenAccountIx;
+    const preUserTokenBalance = ataExists
+      ? await getTokenAccountBalance(this.connection, inputTokenAccount)
```

Comment:
const preUserTokenBalance = inputTokenAccountIx ? "0" : await getTokenAccountBalance(this.connection, inputTokenAccount);